### PR TITLE
DESK-765 connection notifications

### PR DESCRIPTION
**Pull request by JamieRuderman**
### Description

Subscribe to push notifications for device online/offline and connect/disconnect events
Display system notification and update local state

### Test plan

1. Sign in
2. Go to a service and connect
3. You should see the state update and get a connection notice
4. Similarly you should test devices going on/offline and remote connections

### Ticket

<https://remoteit.atlassian.net/browse/DESK-765>

### CODE REVIEW

File: backend/src/ConnectionPool.ts [hunk 1]
```diff
@@ -86,9 +86,15 @@ export default class ConnectionPool {
   set = (connection: IConnection, setCLI?: boolean) => {
     if (!connection) Logger.warn('No connections to set!', { connection })
     let instance = this.find(connection.id)
-    if (instance) instance.set(connection, setCLI)
-    else instance = this.add(connection)
-    this.updated()
+    if (instance) {
+      if (JSON.stringify(connection) !== JSON.stringify(instance.params)) {
```
Comment (mauricio0000) [inline backend/src/ConnectionPool.ts:90]: could you validate that instance is a object?

File: backend/src/sharedCopy/constants.ts [hunk 2]
```diff
@@ -17,9 +17,11 @@ export const CALLBACK_URL =
   process.env.REACT_APP_CALLBACK_URL || process.env.CALLBACK_URL || process.env.NODE_ENV === 'development'
     ? 'https://dev-auth.internal.remote.it/v1/callback'
     : 'https://auth.api.remote.it/v1/callback'
+export const WEBSOCKET_URL = 'wss://ws.remote.it/beta'
```
Comment (mauricio0000) [inline backend/src/sharedCopy/constants.ts:20]: don't need any environtmen variable?

File: frontend/src/models/auth.ts [hunk 3]
```diff
@@ -100,9 +101,10 @@ export default createModel<RootModel>()({
         dispatch.auth.fetchUser()
       }
     },
-    async authenticated(_: void, rootState: any) {
+    async authenticated(_: void, rootState) {
       if (rootState.auth.authenticated) {
         dispatch.auth.setBackendAuthenticated(true)
+        await cloudController.init()
```
Comment (mauricio0000) [inline frontend/src/models/auth.ts:107]: could you make any promise all with the await? I mean 4 await in the same method is not good

File: frontend/src/models/devices.ts [hunk 4]
```diff
@@ -243,15 +242,17 @@ export function selectService(state: ApplicationState, serviceId: string) {
...
+  let service: IService | undefined
+  const device = devices.find(
+    d =>
+      d.id === id ||
+      d?.services?.find(s => {
+        if (s.id === id) {
+          service = s
+          return true
+        }
+        return false
```
Comment (mauricio0000) [inline frontend/src/models/devices.ts:254]: why you need this `return false`?

File: backend/src/ConnectionPool.ts (hunk 1 above)
Comment (JamieRuderman) [reply backend/src/ConnectionPool.ts:90]: In this case it doesn't really matter if it's an object or not - just if we've found it in the collection.

File: backend/src/sharedCopy/constants.ts (hunk 2 above)
Comment (JamieRuderman) [reply backend/src/sharedCopy/constants.ts:20]: Not yet anyway - we can add one if needed - we'll probably also need a production url as well at some point.

File: frontend/src/models/devices.ts (hunk 4 above)
Comment (JamieRuderman) [reply frontend/src/models/devices.ts:254]: Probably dont' need - I'll remove.

File: frontend/src/models/auth.ts (hunk 3 above)
Comment (JamieRuderman) [reply frontend/src/models/auth.ts:107]: I know some of those require they have data before the others. But I don't need the await in the case of the cloudController init, so I'll remove it.

**Pull request merged**